### DISCUSSION TITLE
Minimize the size of the release version

### DISCUSSION THREAD

**Discussion by a1994846931931** [eclipse-theia/theia#8039] [General]
## Question
I was trying to minimize the release version and although managed to reduce the size by building in production mode, the package is still very large. So I'd like to know if there is some general way to reduce the size of the package? Besides, I have the following questions particularly:

1. the root `package.json` have `typescript` in devDependency (and somehow `yarn` won't delete it even in production mode), which accounts for 53MB disk size in `node_modules`, and I wonder if there's a better way to get rid of it except deleting it manually after the build?
2. some dependencies of dev-packages are very large and I wonder if they are necessary. For example, after deleting `webpack-cli` from `application-manager`'s dependency, the total size of the node_modules reduced by about 40MB;
3. I don't need electron version of theia, so I deleted `dev-packages/electron` and `examples/electron` before the build, but you can tell that there are still some electron-related dependencies remain in node_modules, for example, electron-rebuild, so is there a way to remove electron thoroughly?

### My Build Script
```
yarn --ignore-engines --link-duplicates
yarn --ignore-engines --link-duplicates --production
yarn autoclean --force
rm -fr node_modules/electron/dist
modclean -r
find ./packages -name "*.ts" -exec rm {} \+
find ./dev-packages -name "*.ts" -exec rm {} \+
find ./examples -name "*.ts" -exec rm {} \+
```

**Comment by akosyakov** [comment]
Please see the example of minimal docker distribution for Web development: https://github.com/theia-ide/theia-apps/blob/master/theia-docker/Dockerfile#L8-L17
